### ISSUE TITLE
Examples And Performance Results

### ISSUE THREAD

**Issue by keightyfive** [facebookresearch/TensorComprehensions#347]
Hi,

I want to run the kernels and production models from the paper (https://arxiv.org/pdf/1802.04730.pdf), but can't find the repo on Github.
Do you have the implementations along with corresponding test data and .sh scripts available anywhere?

Cheers
Kevin

**Comment by ftynse** [comment]
> but can't find the repo on Github

Well, you have found it.

As the paper indicates, those results were based on a preliminary version of TC.  The released version has many different components and may perform better (or sometimes worse) on various benchmarks. 

Please take a look at `./test`, `./tc/examples` and `./tc/benchmarks` in the current tree.

**Comment by keightyfive** [comment]
Okay thanks, bit of a blind deer today.
I saw ./test but thought it was more for testing mapping or more general stuff...

**Comment by nicolasvasilache** [comment]
@keightyfive test is just unit tests indeed.
For the paper you want to look at ```./tc/benchmarks```.

If you want to run the whole tuner, in ```./tc/benchmarks/scripts``` you'll find scripts that run on an HPC style environment with slurm.
However they may not have been updated following the tree structure change.

But it should be the matter of just doing a sed, contributions most welcome ;)

Also, I recently added a few more things in ```./tc/examples``` but we haven't reported perf results there yet, they are still functional and show some reasonable perf.

**Comment by prigoyal** [comment]
Hi @keightyfive , the benchmark scripts in `./tc/benchmarks/scripts` should also work without requiring any changes. Let us know if this addresses your question and if the issue can be closed. Thanks.

**Comment by keightyfive** [comment]
I'll try setting things up and will try running some of the kernels on a cluster with slurm this week. You can close the thread for now, but I'm sure I'll be back soon :-)

**Comment by keightyfive** [comment]
Hi,
so I had to slightly modify the autotuner_parallel.sh script, I still have a few questions:

1. Does the script work in srun mode as well?
2. What exactly does the script do? Does it tune the mapping for all the benchmarks and then runs all of these 1000 times as described in the paper?
3. Is there a way to run the kernels individually?
4. I installed TC using the conda package with pytorch integration (https://facebookresearch.github.io/TensorComprehensions/framework/pytorch_integration/getting_started.html#installation), since the building from source won't work, which I reported in a separate issue. Will I need the package for caffee2 as well for the benchmarks?

Cheers, Kevin
